doc/user/tutorials/making_your_first_app.rst

```diff
@@ -0,0 +1,510 @@
+perform your own methodology.  Here we just have it make up dummy values representing flux
+and temperatures.
+
+Making the dummy flux kernel
+----------------------------
+In a previous tutorial, we made a function that sets a dummy flux to all parts of the core
+based on a radial distance from the origin. Here we will re-use that code but package it
+more formally so that ARMI can actually run it for us from a user perspective.
+
+The interface is responsible largely for scheduling activities to run at various time
+points. For a flux calculation, we want it to compute at every single time node, so we use
+the :py:meth:`armi.interfaces.Interface.interactEveryNode` hook.
+
+These interaction hooks can call arbitrarily complex code. The code could, for example:
+
+* Run an external executable locally
+* Submit an external code to a cloud HPC and wait for it to complete
+* Run an internal physics tool
+
+Here it just a tiny bit of math locally.
+
+.. code-block:: python
+    :caption: ``myapp/fluxSolver.py``
+
+    import os
+
+    import numpy as np
+
+    from armi import runLog
+    from armi import interfaces
+    from armi.physics import neutronics
+
+
+    class FluxInterface(interfaces.Interface):
+        name = "dummyFlux"
+
+        def interactEveryNode(self, cycle=None, timeNode=None):
+            runLog.info("Computing neutron flux and power.")
+            setFakePower(self.r.core)
+
+
+    def setFakePower(core):
+        midplane = core[0].getHeight()/2.0
+        center = np.array([0,0,midplane])
+        peakPower = 1e6
+        mgFluxBase = np.arange(5)
+        for a in core:
+            for b in a:
+                vol = b.getVolume()
+                coords = b.spatialLocator.getGlobalCoordinates()
+                r = np.linalg.norm(abs(coords-center))
+                fuelFlag = 10 if b.isFuel() else 1.0
+                b.p.power = peakPower / r**2 * fuelFlag
+                b.p.pdens = b.p.power/vol
+                b.p.mgFlux = mgFluxBase*b.p.pdens
+
+
+Making the thermal/hydraulics kernel
+------------------------------------------
+Since we told the ARMI plugin to schedule the flux solver before thermal/hydraulics solver
+via the ``ORDER`` attribute, we can depend on there being up-to-date block-level ``power``
+state data loaded onto the ARMI reactor by the time this thermal/hydraulics solver gets
+called by the ARMI main loop.
+
+We'll make a somewhat meaningful (but still totally academic) flow solver here that uses
+energy conservation to determine an idealized coolant flow rate. To do this it will
+compute the total power produced by each assembly.
+
+.. math::
+
+    q''' = \dot{m} C_p \Delta T
+
+
+.. code-block:: python
+    :caption: ``myapp/thermalSolver.py``
+
+    from armi import interfaces
+    from armi.reactor.flags import Flags
+    from armi import runLog
+
+    # hard coded inlet/outlet temperatures
+    inletInC = 360.0
+    outletInC = 520.0
```

Comment:
It would be a good opportunity to introduce the settings hooks into this to make use of the CS for settings. It will add a bit more complexity, but could be useful to see how a user can reuse the settings for their own needs.

Reply:
I plan on doing that later